stdlib/event/time.lua

```diff
@@ -0,0 +1,73 @@
+--- Time Event module
+-- @module Event.Time
+
+require 'stdlib/event/event'
+
+Event.Time = {}
+Event.Time._last_change = {}
+
+--All times are offset by 0.5
+--This is because both EvoGUI and MoWeather already apply that offset.
+--Following the precedent to remain consistent.
+--Actually, this little snippet is even borrowed from EvoGUI.
+if remote.interfaces.MoWeather then
+    -- assume MoWeather's getdaytime is sane
+    function Event.Time.get_day_time(surface_name_or_index) return remote.call("MoWeather", "getdaytime", surface_name_or_index) end
```

Comment:
I had the same debate with myself decided I'd go with compatibility since it was easy enough to include and would help foster a discussion about if compatibility with existing mods was something worth undertaking or not or to be determined on a case-by-case basis.